pkg/registry/core/pod/storage/storage.go

```diff
@@ -180,20 +181,32 @@ func (r *BindingREST) Create(ctx context.Context, name string, obj runtime.Objec
 		}
 	}
 
-	err = r.assignPod(ctx, binding.Name, binding.Target.Name, binding.Annotations, dryrun.IsDryRun(options.DryRun))
+	err = r.assignPod(ctx, binding.UID, binding.ResourceVersion, binding.Name, binding.Target.Name, binding.Annotations, dryrun.IsDryRun(options.DryRun))
 	out = &metav1.Status{Status: metav1.StatusSuccess}
 	return
 }
 
 // setPodHostAndAnnotations sets the given pod's host to 'machine' if and only if it was
 // previously 'oldMachine' and merges the provided annotations with those of the pod.
 // Returns the current state of the pod, or an error.
-func (r *BindingREST) setPodHostAndAnnotations(ctx context.Context, podID, oldMachine, machine string, annotations map[string]string, dryRun bool) (finalPod *api.Pod, err error) {
+func (r *BindingREST) setPodHostAndAnnotations(ctx context.Context, podUID types.UID, podResourceVersion, podID, oldMachine, machine string, annotations map[string]string, dryRun bool) (finalPod *api.Pod, err error) {
 	podKey, err := r.store.KeyFunc(ctx, podID)
 	if err != nil {
 		return nil, err
 	}
-	err = r.store.Storage.GuaranteedUpdate(ctx, podKey, &api.Pod{}, false, nil, storage.SimpleUpdate(func(obj runtime.Object) (runtime.Object, error) {
+
+	var preconditions *storage.Preconditions
+	if podUID != "" || podResourceVersion != "" {
```

Comment:
why not break this into two blocks:
```
if podUID != "" {
}

if podResourceVersion != "" {
}
```

Reply:
> why not break this into two blocks:
> 
> ```
> if podUID != "" {
> }
> 
> if podResourceVersion != "" {
> }
> ```

do you mean two blocks like this ?
```
var preconditions *storage.Preconditions
if podUID != "" {
	preconditions = &storage.Preconditions{UID: &podUID}
}

if podResourceVersion != "" {
	if preconditions == nil {
		preconditions = &storage.Preconditions{ResourceVersion: &podResourceVersion}
	} else {
		preconditions.ResourceVersion = &podResourceVersion
	}
}
```

I think the code above looks asymmetric  :-)

Reply:
sounds good, forget my suggestion.